File: app/src/main/java/com/semanientreprise/soundrecorderbox/Recordings.java
package com.semanientreprise.soundrecorderbox;

import android.os.Parcel;
import android.os.Parcelable;

import io.objectbox.annotation.Entity;
import io.objectbox.annotation.Id;
import io.objectbox.annotation.NameInDb;

@Entity
public class Recordings implements Parcelable {

    @Id
    public long id;

    @NameInDb("RecordingName")
    private String recording_name;
    @NameInDb("RecordingPath")
    private String recording_path;
    @NameInDb("RecordingLength")
    private long recording_length;
    @NameInDb("RecordingTime")
    private long recording_time_added;

    public Recordings() {}


    public Recordings(long id, String recording_name, String recording_path,
                      long recording_length) {
        this.id = id;
        this.recording_name = recording_name;
        this.recording_path = recording_path;
        this.recording_length = recording_length;
        this.recording_time_added = System.currentTimeMillis();
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getRecording_name() {
        return recording_name;
    }

    public void setRecording_name(String recording_name) {
        this.recording_name = recording_name;
    }

    public String getRecording_path() {
        return recording_path;
    }

    public void setRecording_path(String recording_path) {
        this.recording_path = recording_path;
    }

    public long getRecording_length() {
        return recording_length;
    }

    public void setRecording_length(int recording_length) {
        this.recording_length = recording_length;
    }

    public long getRecording_time_added() {
        return recording_time_added;
    }

    public void setRecording_time_added(int recording_time_added) {
        this.recording_time_added = recording_time_added;
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeLong(this.id);
        dest.writeString(this.recording_name);
        dest.writeString(this.recording_path);
        dest.writeLong(this.recording_length);
        dest.writeLong(this.recording_time_added);
    }

    protected Recordings(Parcel in) {
        this.id = in.readLong();
        this.recording_name = in.readString();
        this.recording_path = in.readString();
        this.recording_length = in.readLong();
        this.recording_time_added = in.readLong();
    }

    public static final Parcelable.Creator<Recordings> CREATOR = new Parcelable.Creator<Recordings>() {
        @Override
        public Recordings createFromParcel(Parcel source) {
            return new Recordings(source);
        }

        @Override
        public Recordings[] newArray(int size) {
            return new Recordings[size];
        }
    };
}

File: app/src/main/java/com/semanientreprise/soundrecorderbox/Recordings.kt
package com.semanientreprise.soundrecorderbox

import android.os.Parcel
import android.os.Parcelable
import io.objectbox.annotation.Entity
import io.objectbox.annotation.Id
import io.objectbox.annotation.NameInDb

@Entity
class Recordings : Parcelable {
    @JvmField
    @Id
    var id: Long = 0
    @NameInDb("RecordingName")
    var recording_name: String? = null
    @NameInDb("RecordingPath")
    var recording_path: String? = null
    @NameInDb("RecordingLength")
    var recording_length: Long = 0
        private set
    @NameInDb("RecordingTime")
    var recording_time_added: Long = 0
        private set

    constructor() {}
    constructor(id: Long, recording_name: String?, recording_path: String?,
                recording_length: Long) {
        this.id = id
        this.recording_name = recording_name
        this.recording_path = recording_path
        this.recording_length = recording_length
        recording_time_added = System.currentTimeMillis()
    }

    fun setRecording_length(recording_length: Int) {
        this.recording_length = recording_length.toLong()
    }

    fun setRecording_time_added(recording_time_added: Int) {
        this.recording_time_added = recording_time_added.toLong()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeLong(id)
        dest.writeString(recording_name)
        dest.writeString(recording_path)
        dest.writeLong(recording_length)
        dest.writeLong(recording_time_added)
    }

    protected constructor(`in`: Parcel) {
        id = `in`.readLong()
        recording_name = `in`.readString()
        recording_path = `in`.readString()
        recording_length = `in`.readLong()
        recording_time_added = `in`.readLong()
    }

    companion object {
        val CREATOR: Parcelable.Creator<Recordings> = object : Parcelable.Creator<Recordings?> {
            override fun createFromParcel(source: Parcel): Recordings? {
                return Recordings(source)
            }

            override fun newArray(size: Int): Array<Recordings?> {
                return arrayOfNulls(size)
            }
        }
    }
}